src/rgw/rgw_op.cc

```diff
@@ -5035,13 +5035,11 @@ void RGWPutACLs::execute()
     if (op_ret < 0)
       return;
 
-    new_data = strdup(ss.str().c_str());
-    free(data);
-    data = new_data;
-    len = ss.str().size();
+    data.clear();
```

Comment:
is this right?  a new buffer::list has no data, but if we're appending to an existing one, we do not want to clear

Reply:
Just trying to re-use the bufferlist (data) available here (the way data as a char* was re-used), we need to overwrite the existing data with the new data, that is why used clear().

Reply:
I don't think it buys us much, since we don't re-use memory or buffer::ptrs or anything, but it doesn't hurt anything either.